lib/AArch64Base.ml

```diff
@@ -1976,6 +1978,24 @@ let get_next =
   | I_LDXP _|I_STXP _|I_UDF _
     -> [Label.Next;]
 
+(* Check instruction validity, beyond parsing *)
+let mask12 = (1 lsl 12)-1
+
+let is_12bits k = k land mask12 = k
+
+let is_valid i =
+  match i with
+  | I_OP3 (_,(ADD|SUB|ADDS|SUBS),_,ZR,K _,_)
+  | I_OP3 (_,(ADD|SUB),ZR,_,K _,_)
+    -> false
+  | I_OP3 (_,(ADD|SUB|ADDS|SUBS),_,_,K k,(S_NOEXT|S_LSL (0|12)))
+    -> is_12bits (abs k)
```

Comment:
If I understand correctly, the Arm ARM says that it should be unsigned? Is that because later on, `ADD` of a negative immediate will be converted in a `SUB`? Or do we want to remove the `abs` here?

Reply:
Your understanding is correct: using either ADD or SUB the immediate constant size is 12 bits positive + sign.